ignite/contrib/engines/common.py

```diff
@@ -75,9 +81,11 @@ def setup_common_training_handlers(
         with_pbars=with_pbars,
         with_pbar_on_iters=with_pbar_on_iters,
         log_every_iters=log_every_iters,
-        device=device,
+        stop_on_nan=stop_on_nan,
     )
-    if dist.is_available() and dist.is_initialized():
+    from ignite.distributed.utils import _SerialModel
+
+    if idist.model_name() != _SerialModel.name:
```

Comment:
Maybe, this better should be : 
```suggestion
    if idist.get_world_size() > 1:
```